src/cpp/src/group_beam_searcher.cpp

```diff
@@ -412,20 +419,18 @@ EncodedResults beam_search(ov::InferRequest& lm,
         }
         size_t batch_size = next_tokens.size();
         // Set pointers
-        ov::Tensor next_tokens_tensor(ov::element::i64, {batch_size, 1});
-        std::memcpy(next_tokens_tensor.data(), next_tokens.data(), next_tokens_tensor.get_byte_size());
-        lm.set_tensor("input_ids", next_tokens_tensor);
-
-        ov::Tensor next_beams_tensor(ov::element::i32, {batch_size});
-        std::memcpy(next_beams_tensor.data(), next_beams.data(), next_beams_tensor.get_byte_size());
-        lm.set_tensor("beam_idx", next_beams_tensor);
+        lm.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
+        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
 
         // Set auxiliary inputs
         update_attention_mask_with_beams(lm.get_tensor("attention_mask"), next_beams);
         if (position_ids_available)
             update_position_ids(lm.get_tensor("position_ids"), lm.get_tensor("attention_mask"));
     }
 
+    //reset all inputs with empty tensors
+    reset_inputs(lm);
```

Comment:
Rename the function to `reset_all_inputs_to_empty_tensors()` so you don't need the comment.

Reply:
done